go/vt/vtgate/executor_scatter_stats.go

```diff
@@ -89,9 +89,13 @@ func (e *Executor) gatherScatterStats() (statsResults, error) {
 	resultItems := make([]*statsResultItem, len(plans))
 	for i, plan := range plans {
 		route := routes[i]
+		var avgTimePerQuery int64 = 0
+		if plan.ExecCount != 0 {
+			avgTimePerQuery = plan.ExecTime.Nanoseconds() / int64(plan.ExecCount)
+		}
 		resultItems[i] = &statsResultItem{
 			Query:                  plan.Original,
-			AvgTimePerQuery:        time.Duration(plan.ExecTime.Nanoseconds() / int64(plan.ExecCount)),
+			AvgTimePerQuery:        time.Duration(avgTimePerQuery),
 			PercentTimeOfReads:     float64(100 * plan.ExecTime / readOnlyTime),
```

Comment:
You still need to convert all these integral denominators to floats to prevent divide-by-zero panic()s, as per the comment in the closed PR

Creating new PRs for each round of code review makes it trickier for people to keep track of the conversation and previous comments. In case you weren't aware, force-pushing to your existing branch on github will update an existing PR.

Reply:
Thanks, am kind of new to use Github system. Messed up a bit so created a newer branch. Let me update the code according to comment

Reply:
Cast the dominators to float64